homeassistant/components/samsungtv/__init__.py

```diff
@@ -290,6 +291,20 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         version = 2
         hass.config_entries.async_update_entry(config_entry, version=2)
 
-    LOGGER.debug("Migration to version %s successful", version)
+    if version == 2:
+        if minor_version < 2:
+            # Cleanup invalid MAC addresses - see #103512
+            dev_reg = dr.async_get(hass)
+            for device in dr.async_entries_for_config_entry(
+                dev_reg, config_entry.entry_id
+            ):
+                for connection in device.connections:
+                    if connection == (dr.CONNECTION_NETWORK_MAC, "none"):
+                        dev_reg.async_remove_device(device.id)
```

Comment:
It's safer to remove the config entry id from the device. The device will be removed automatically if it was the only config entry for the device.

Reply:
It has been refactored in https://github.com/home-assistant/core/pull/118556